### DISCUSSION TITLE
when using IVF_PQ, why nlist and nbits are both needed?

### DISCUSSION THREAD

**Discussion by hanj99** [milvus-io/milvus#21186] [Q&A and General discussion] [answered]
Hi. I'm curious about the needs for the nbits in IVF_PQ.

As far as i know, If i specify the nlist=256 then each subspace has 256 centroids and automatically the bits required to represent 256 codes is 8bits (which i think equals to nbits)
So my question is why nbits parameter exist in creating index.

Isn't it sufficient just specify nlist when using IVF_PQ?
Thank you.

**Comment by xiaofan-luan** [comment, marked as answer]
Hi @hanj99 
From basic explain, nlist is related to IVF and nbit is related to PQ.

nlist + nprobe decide the recall/performance tradeoff.
nbit decide  the recall/memory tradeoff

**Reply by hanj99** [reply]
Thank you @xiaofan-luan.

Maybe I'm confused between the IVF centroids and PQ centroids.
